sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuBroadcastToCpuExec.scala

```diff
@@ -83,36 +87,27 @@ case class GpuBroadcastToCpuExec(override val mode: BroadcastMode, child: SparkP
             }
 
             // deserialize to host buffers in the driver and then convert to rows
-            val dataTypes = child.output.map(_.dataType)
-
-            val gpuBatches = serializedBatches.safeMap { cb =>
-              val hostColumns = (0 until cb.header.getNumColumns).map { i =>
-                val columnHeader = cb.header.getColumnHeader(i)
-                val hcv = new HostColumnVector(
-                  columnHeader.getType,
-                  columnHeader.rowCount,
-                  Optional.of(columnHeader.nullCount),
-                  cb.buffer,
-                  null, null, List.empty.asJava)
-                new RapidsHostColumnVector(dataTypes(i), hcv)
+            val gpuBatches = withResource(serializedBatches) { _ =>
+              serializedBatches.safeMap { cb =>
+                val hostColumns = GpuBroadcastToCpuExec.buildHostColumns(cb.header, cb.buffer,
```

Comment:
follow up for later: not sure if there is an easy way to unit test or specifically test this since we obviously building this wrong, or perhaps if we get some dpp/aqe testing with data validation it would cover it.

Reply:
I think DPP+AQE testing would cover it.  Filed #3724.